doc-hints-for-docker: gitconfig for private repos. Fixes #4539 (#4564)
oss-review-toolkit/ort: docs/hints-for-use-with-docker.md

   ```

   **Important:** Ensure that the `.netrc` file has been created at `netrc_folder_path` before running the command, otherwise Docker will create a folder `.netrc` inside the container, instead of mounting a single file.

> It is also possible to use ```git config``` to setup access to private repositories, such as github repos.

Author: Thanks for the feedback. Changed it as suggested.

> 1. Configure your gitconfig with your github credentials

Reviewer: Instead of "gitconfig", this should say "`.gitconfig`" as below. Also "github" should be spelled "GitHub".

Reviewer: Sorry, my quoting was probably not clear:

    `.github`

Should use backticks in the Markdown file.

Author: "gitconfig" is not written `.gitconfig`
"github" is now spelled "GitHub".

    ```shell
    [url "https://<username>:<password>@github.com"]

> 	         insteadOf = /https://github.com    

Reviewer 2: I think the leading `/` in the URL is incorrect.

Author: Removed it.

>     ```

Author: Thanx, changed it as suggested.

> When using the container, the .gitconfig can be mounted into the container:

Author: Thanx, changed it as you suggested.

    ```shell

>     docker run -e -v $(pwd):/project -v ~/.gitconfig:/root/.gitconfig ort --info analyze -f JSON -i /project -o /project/ort/analyzer

Author: Thanx, changed it as suggested.

   ```

   **Important:** Ensure that the `.netrc` file has been created at `netrc_folder_path` before running the command, otherwise Docker will create a folder `.netrc` inside the container, instead of mounting a single file.

It is also possible to use `git config` to setup access to private repositories.

> 1. Configure your gitconfig with your github credentials
> 
>     ```shell
>     git config --global url."https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com".insteadOf /"https://github.com"
>     ```
> 
>     ```shell
>     cat ~/.gitconfig
>     [url "https://<username>:<password>@github.com"]
> 	         insteadOf = /https://github.com    
>     ```

Reviewer 2: I think we should not document pure Git behaviour in this documentation.
A user does not need to run the `git config` command to create the .gitconfig file, they could also create it manually.

````suggestion
1. Configure the `.gitconfig` with the repository credentials. E.g. Github:
    ```
    [url "https://<username>:<password>@github.com"]
             insteadOf = https://github.com    
    ```
````

Author: Thanx, implemented changes as suggested.
